library/patient.inc

```diff
@@ -812,6 +814,15 @@ function getPatientPID($args)
 
     return $returnval;
 }
+function getPatientDoBonly($pid)
+{
+    if (empty($pid)) {
```

Comment:
You stated in the function block that `$pid` is required, without it this function cannot, well, function. Instead of checking for empty why not break that out to `validatePID()` (One may already exist). This function would just return a boolean. Why abstract it to another function? `pid` is a number right now, but a day will come when it may not be just a number. You can make `validatePID()` assess to see if it's a `int` (Or whatever). Basically the purpose of that function is to say "Hey, I don't care _what_ this `pid` actually is, can you just tell me if what I'm passing you _could_ be a `pid`.

Reply:
On this one, the intent is to make sure that the function is not being called with an empty value for $pid. The PID is already established if I am reading your critique right so I don't understand the need to validate it if I know at this point it already exists.  I don't care what the PID is made up of, I just care that the variable $pid is not empty.